### DISCUSSION TITLE
Why is axios not allowing to attach a File object in the PUT method body?

### DISCUSSION THREAD

**Discussion by krishmarsonia** [axios/axios#6493] [General]
I am working on a next.js project, where I have to upload an image/video in the S3 bucket through Pre-signed URL. In every video/blog/article everyone is saying to append the file in the body. But, I am unable to do so, as axios is giving me an error of **Data after transformation must be a string, an ArrayBuffer, a Buffer, or a Stream**, I have done many things to resolve this issue, I have created ArrayBuffers, I have created streams of the file, but nothing was of any use. Then I tried it with fetch API and it worked magically in the first time only. Is there any solution for this problem?

**Comment by suhaotian** [comment]
If you want to upload file try use `FormData`:

```ts
const form = new FormData();
form.append('file', FileObject);

axios.put(form);
```

**Reply by krishmarsonia** [reply]
> If you want to upload file try use `FormData`:
> 
> ```ts
> const form = new FormData();
> form.append('file', FileObject);
> 
> axios.put(form);
> ```

I have tried that, but in the FormData we have to pass the data in key value pair, so when I had tried to upload the FormData in the pre-signed URL then it gets uploaded, but it does not reflect the data(image/video). what I mean is after uploading the data when I try to open the image/video it shows nothing. It shows something like this.

![image](https://github.com/user-attachments/assets/47b8175f-9404-4af9-bba8-e2b9ce3827c1)

**Reply by suhaotian** [reply]
> > If you want to upload file try use `FormData`:
> > ```ts
> > const form = new FormData();
> > form.append('file', FileObject);
> > 
> > axios.put(form);
> > ```
> 
> I have tried that, but in the FormData we have to pass the data in key value pair, so when I had tried to upload the FormData in the pre-signed URL then it gets uploaded, but it does not reflect the data(image/video). what I mean is after uploading the data when I try to open the image/video it shows nothing. It shows something like this.
> 
> ![image](https://private-user-images.githubusercontent.com/69512823/351973289-47b8175f-9404-4af9-bba8-e2b9ce3827c1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE4ODU1NzUsIm5iZiI6MTcyMTg4NTI3NSwicGF0aCI6Ii82OTUxMjgyMy8zNTE5NzMyODktNDdiODE3NWYtOTQwNC00YWY5LWJiYTgtZTJiOWNlMzgyN2MxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI1VDA1Mjc1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEzMmNhZDg1MTNjYjZiNjU4MWY1YjkwMTg2ZWMwMDJhNjg1MzA1ODgxYTA1NjdhMTE0YTA4MDU5NTI4Y2YxNWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.LIbAu3AO36BlbXuWCI45Hs7kTX9a6hCkvCKTYMFzavU)

Can you show your full upload example code?
